pkg/cfn/builder/iam.go

```diff
@@ -132,5 +132,18 @@ func (n *NodeGroupResourceSet) addResourcesForIAM() {
 		},
 	)
 
+	if n.spec.Addons.WithIAM.AutoScalingGroup {
+		n.rs.attachAllowPolicy("AutoScalingControl", refIR, "*",
```

Comment:
let's keep naming consistent with the other inline policy:
```suggestion
		n.rs.attachAllowPolicy("PolicyAutoScaling", refIR, "*",
```